lib/Intlc/Linter.hs

```diff
@@ -0,0 +1,37 @@
+module Intlc.Linter where
+
+import           Intlc.ICU
+import           Prelude   hiding (Type)
+
+data Status
+  = Success
+  | Failure Text
+  deriving (Eq, Show)
+
+countInterpolations :: NEStream -> Int
+countInterpolations = foldr go 0
+  where
+    count = \case
+      String      -> 0
+      Number      -> 0
+      Date {}     -> 0
+      Time {}     -> 0
+      PluralRef   -> 0
+      Bool {}     -> 1
+      Plural {}   -> 1
+      Select {}   -> 1
+      Callback {} -> 1
+    go :: Token -> Int -> Int
+    go t n = case t of
+      Plaintext {}              -> n
+      Interpolation ((Arg _ x)) -> count x + n
+
+lint :: Message -> Status
+lint m = case m of
+  Static {}        -> Success
+  Dynamic neStream -> (mkStatus . countInterpolations) neStream
+    where
+      mkStatus n = if n > 1
```

Comment:
This is a perfect use case for guards, which essentially let you pattern match on expressions utilising function arguments. To find the right thing search for `otherwise`.

Reply:
e601402d4960b69cafe68e8eb0360dcc00a298ed